test/integration/failover_stateful_test.lua

```diff
@@ -132,107 +246,103 @@ function g.test_stateboard_restart()
+        t.assert_equals(helpers.list_cluster_issues(R1), {})
     end)
 
     helpers.retrying({}, function()
-        t.assert_equals(eval('router',    q_leadership), storage_1_uuid)
-        t.assert_equals(eval('storage-1', q_leadership), storage_1_uuid)
-        t.assert_equals(eval('storage-2', q_leadership), storage_1_uuid)
-        t.assert_equals(eval('storage-3', q_leadership), storage_1_uuid)
+        t.assert_equals(R1.net_box:eval(q_leadership), storage_1_uuid)
+        t.assert_equals(S1.net_box:eval(q_leadership), storage_1_uuid)
+        t.assert_equals(S2.net_box:eval(q_leadership), storage_1_uuid)
+        t.assert_equals(S3.net_box:eval(q_leadership), storage_1_uuid)
     end)
 
-    t.assert_equals(eval('storage-1', q_readonliness), false)
-    t.assert_equals(eval('storage-2', q_readonliness), true)
-    t.assert_equals(eval('storage-3', q_readonliness), true)
-end
+    t.assert_equals(S1.net_box:eval(q_readonliness), false)
+    t.assert_equals(S2.net_box:eval(q_readonliness), true)
+    t.assert_equals(S3.net_box:eval(q_readonliness), true)
+end)
 
-function g.test_coordinator_restart()
-    eval('router', [[
+add('test_coordinator_restart', function(g)
+    R1.net_box:eval([[
         local cartridge = require('cartridge')
         local coordinator = cartridge.service_get('failover-coordinator')
         return coordinator.stop()
     ]])
 
     helpers.retrying({}, function()
-        t.assert_equals(
-            g.stateboard.net_box:call('get_coordinator'),
-            nil
-        )
+        t.assert_equals(g.client:get_session():get_coordinator(), nil)
     end)
 
-    eval('router', [[
+    R1.net_box:eval([[
         local confapplier = require('cartridge.confapplier')
+        local state = confapplier.wish_state('RolesConfigured')
+        assert(state == 'RolesConfigured', state)
         return confapplier.apply_config(confapplier.get_active_config())
     ]])
 
     helpers.retrying({}, function()
         t.assert_equals(
-            g.stateboard.net_box:call('get_coordinator'),
+            g.client:get_session():get_coordinator(),
             {
-                uri = g.cluster:server('router').advertise_uri,
-                uuid = g.cluster:server('router').instance_uuid,
+                uri = R1.advertise_uri,
+                uuid = R1.instance_uuid,
             }
         )
+        t.assert_equals(helpers.list_cluster_issues(R1), {})
     end)
-end
-
-function g.test_leader_restart()
-    t.assert_equals(
-        g.stateboard.net_box:call('longpoll', {0}),
-        {
-            [router_uuid] = router_1_uuid,
-            [storage_uuid] = storage_1_uuid,
-        }
-    )
+end)
 
+add('test_leader_restart', function(g)
     -----------------------------------------------------
-    g.cluster:server('storage-1'):stop()
-    t.assert_equals(
-        g.stateboard.net_box:call('longpoll', {3}),
+    g.client:longpoll(0)
```

Comment:
Why did you add longpoll(0) here?

Reply:
It's already been there. It's purpose is to abandon all previous events and act as a sync point of S1 death.